src/caffe/layers/image_data_layer.cpp

```diff
@@ -37,10 +37,13 @@ void ImageDataLayer<Dtype>::DataLayerSetUp(const vector<Blob<Dtype>*>& bottom,
   const string& source = this->layer_param_.image_data_param().source();
   LOG(INFO) << "Opening file " << source;
   std::ifstream infile(source.c_str());
-  string filename;
+  std::string line;
+  size_t pos;
   int label;
-  while (infile >> filename >> label) {
-    lines_.push_back(std::make_pair(filename, label));
+  while (getline(infile, line)) {
```

Comment:
Ah, it's argument-dependent lookup. The explicit version (as it is now) seems reasonable to me, especially since we don't use `getline` frequently in Caffe.